lua/tardis/screens/sh_destination.lua

```diff
@@ -7,27 +7,215 @@ TARDIS:AddControl("destination",{
+
+
+	local pitch = vgui.Create("DTextEntry",frame)
+	pitch:SetPlaceholderText("Pitch")
+	pitch:SetPos(btnx*0.97,btny*7)
+	local yaw = vgui.Create("DTextEntry",frame)
+	yaw:SetPlaceholderText("Yaw")
+	yaw:SetPos(btnx*1.08,btny*7)
+	local roll = vgui.Create("DTextEntry",frame)
+	roll:SetPlaceholderText("Roll")
+	roll:SetPos(btnx*1.19,btny*7)
+
+	x:SetNumeric(true)
+	y:SetNumeric(true)
+	z:SetNumeric(true)
+	pitch:SetNumeric(true)
+	yaw:SetNumeric(true)
+	roll:SetNumeric(true)
+
+	local function updatetextinputs(pos,ang)
+		pitch:SetText(ang.p)
+		yaw:SetText(ang.y)
+		roll:SetText(ang.r)
+		x:SetText(pos.x)
+		y:SetText(pos.y)
+		z:SetText(pos.z)
+	end
+	local function fetchtextinputs()
+		local pos = 0
+		local ang = 0
+		if x:GetText() ~= "" and y:GetText() ~= "" and z:GetText() ~= "" then
+			pos = Vector(x:GetText() or 0, y:GetText()or 0, z:GetText() or 0)
+		end
+		if pitch:GetText() ~= "" and yaw:GetText() ~= "" and roll:GetText() ~= "" then
+			ang = Angle(pitch:GetText() or 0, yaw:GetText() or 0, roll:GetText() or 0)
+		end
+		if pos ~= 0 and ang ~= 0 then
+			return pos,ang
+		else
+			return false,false
+		end
+	end
+	if ext:GetData("demat-pos") and ext:GetData("demat-ang") then
+		updatetextinputs(ext:GetData("demat-pos"), ext:GetData("demat-ang"))
+	end
+
+
+
+	local list = vgui.Create("DListView",frame)
+	list:SetSize( frame:GetWide()*0.7, frame:GetTall()*0.95 )
+	list:SetPos( frame:GetWide()*0.26 - list:GetWide()*0.35, frame:GetTall()*0.5 - list:GetTall()*0.5 )
+	list:AddColumn("Name")
+	local map = game.GetMap()
+	local function updatelist()
+		list:Clear()
+		if TARDIS.Locations[map] ~= nil then
+			for k,v in pairs(TARDIS.Locations[map]) do
+				list:AddLine(v.name)
+			end
+		end
+	end
+	updatelist()
+	function list:OnRowSelected(i,row)
+		pos = TARDIS.Locations[map][i].pos
+		ang = TARDIS.Locations[map][i].ang
+		updatetextinputs(pos,ang)
+	end
+
+	local new = vgui.Create("DButton", frame)
+	new:SetSize( frame:GetWide()*0.07, frame:GetTall()*0.1 )
+	new:SetPos(pitch:GetPos()+5,frame:GetTall()*0.35 - button:GetTall()*0.5)
+	new:SetText("New")
+	new:SetFont("TARDIS-Default")
+	function new:DoClick()
+		local name = ""
+		local pos = Vector(0,0,0)
+		local ang = Angle(0,0,0)
+		Derma_StringRequest("New Location",
+		"Enter the name for the new location",
+		"New Location", 
+		function(text)
+			 name = text
+			 Derma_Query("Use current TARDIS position and rotation?","New Location","Yes",
+				function() pos = ext:GetPos() ang = ext:GetAngles() TARDIS:AddLocation(pos,ang,name,map) updatelist() end,
+			"No", function() TARDIS:AddLocation(pos,ang,name,map) updatelist() end)
+		end)
+	end
+	local pendingchanges = false
+	local edit = vgui.Create("DButton", frame)
+	edit:SetSize( frame:GetWide()*0.07, frame:GetTall()*0.1 )
+	edit:SetPos(yaw:GetPos()+5,frame:GetTall()*0.35 - button:GetTall()*0.5)
+	edit:SetText("Edit")
+	edit:SetFont("TARDIS-Default")
+	function edit:DoClick()
+		Derma_Message("TODO", "sorry mate", "k")
```

Comment:
pls hide/complete this

Reply:
I'm working on it sorry lol

Reply:
Hid the message for now. I'll complete this soon.